_resources/port1.0/group/python-1.0.tcl

```diff
@@ -55,7 +55,11 @@ default python.consistent_destroot yes
 
 proc python_get_version {} {
     if {[string match py-* [option name]]} {
-        return [string range [option subport] 2 3]
+        if {[string match pypy* [option subport]]} {
+            return [string range [option subport] 0 [string first "-" [option subport]]-1]
+        } else {
+            return [string range [option subport] 2 3]
+        }
```

Comment:
It's a bit weird treating the "pypy" string as a version, but it's hard to think of a better way of handling it without adding significant complexity.